core/src/main/scala/io/projectglow/sql/expressions/PerSampleSummaryStatistics.scala

```diff
@@ -54,38 +54,52 @@ case class PerSampleSummaryStatistics(
 
   override def children: Seq[Expression] = Seq(genotypes)
   override def nullable: Boolean = false
-  override def dataType: DataType = MapType(StringType, MomentAggState.schema)
+
+  override def dataType: DataType =
+    if (optionalFieldIndices(0) != -1) {
+      ArrayType(MomentAggState.schema.add(VariantSchemas.sampleIdField))
```

Comment:
OOC, why do we prepend sample ID here but append for the call summary stats?

Reply:
Good point, better to be consistent. I'll make it the first spot in each.